test/module/irohad/simulator/simulator_test.cpp

```diff
@@ -298,8 +308,8 @@ TEST_F(SimulatorTest, RightNumberOfFailedTxs) {
 
   EXPECT_CALL(*query, getTopBlockHeight()).WillOnce(Return(2));
 
-  EXPECT_CALL(*validator, validate(_, _))
-      .WillOnce(Return(std::make_pair(verified_proposal, tx_errors)));
+  EXPECT_CALL(*validator, validateProxy(_, _))
+      .WillOnce(Return(verified_proposal_and_errors));
```

Comment:
can't we use invoke here and return unique ptr instead of using proxy?

Reply:
Thanks, it worked.